opendevin/runtime/plugins/agent_skills/agentskills.py

```diff
@@ -516,6 +521,30 @@ def _edit_file_impl(
                 f.writelines(lines)
 
             lint_error, first_error_line = _lint_file(file_name)
+            
+            # Select the errors caused by the modification
+            def extract_last_part(line):
+                parts = line.split(':')
+                if len(parts) > 1:
+                    return parts[-1].strip()
+                return line.strip()
+
+            def subtract_strings(str1, str2) -> str:
+                lines1 = str1.splitlines()
+                lines2 = str2.splitlines()
+
+                last_parts1 = [extract_last_part(line) for line in lines1]
+
+                remaining_lines = [line for line in lines2 if extract_last_part(line) not in last_parts1]
```

Comment:
I suspect if this is enough. You are essentially checking the error message and assuming they are unique. That doesn't seem to always hold, although your original example on Slack follows this pattern.

Moreover, you are assuming if an error message appear in both, then it's not a "new error". That doesn't seem right either.

IMO the only reliable way is to compare the line number (or even including column number, if we want to make it more accurate, but line number seems okay). You need to be very careful when you try to map the line numbers before and after the edit.

Reply:
Rethinking about it, I think a better way is to check the line numbers from `lint_error` and exclude those out-of-context based on line numbers. No need to do a linting before edit.

Say you have L1,2,3,4, and you replace L3 with 2 new lines, and then you have L1,2,*3,4*,5, with L3 & L4 being new. We should only retain linting errors for L3-5 (inclusive).